### PR TITLE
Fix Mac build to use UnixName Custom Build Task

### PR DESCRIPTION
**Pull request by chrisvire**
_system_name and _system_type are not available by default in the
environment.  Use new UnixName Custom Build Task to conditionally
build Mac unit tests instead of depending on _system_name.

Define new build/platform.targets that can be used to create
system-specific definitions for conditional compilation.

### CODE REVIEW

File: Palaso.Tests/PlatformUtilities/PlatformTests.cs [hunk 1]
```diff
@@ -4,23 +4,19 @@
...
-// 1) include the following PropertyGroup in the .csproj after all of the DefineConstants
-//    elements.  [Note: _system_name and _system_type are defined by xbuild in later 
-//    versions of Mono.  We tested with Mono 3.12 that comes with Xamarin Studio for 
-//    Mac.  To verify, add /verbosity:diag to xbuild command-line. YMMV.]
+// For code shipped with products, please try to use runtime checks to select 
+// code to run.  There are cases that require compile-time conditionals 
+// (e.g. when testing the runtime or referencing platform specific imported 
+// assemblies like MonoMac.dll).  In this case, do The following:
+// 
+// 1) Include the following at the bollow of the csproj file
```
Comment (ermshiperete) [inline Palaso.Tests/PlatformUtilities/PlatformTests.cs]: bollow?

Comment (chrisvire) [inline Palaso.Tests/PlatformUtilities/PlatformTests.cs]: sigh... bottom.  Not sure how that happened.

Comment (chrisvire) [inline Palaso.Tests/PlatformUtilities/PlatformTests.cs]: Actually, this shouldn't be here anymore.  It is documented in platform.targets.

File: build/platform.targets [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- 
+Include this file at the end of a csproj file to add system-specific defines 
+for conditional compilation. Example: libpalaso/Palaso.Tests/Palaso.Tests.csproj
+These constants will be defined:
+SYSTEM_UNIX - should be compiled on Linux and Mac
+SYSTEM_MAC - should only be compiled on Mac (e.g. Unit Tests specific to Mac)
+SYSTEM_Linix - should only be compiled on Linux
```
Comment (chrisvire) [inline build/platform.targets]: Should be SYSTEM_LINUX

**Pull request merged**